store/__tests__/playbackManager.spec.ts

```diff
@@ -52,721 +52,745 @@ beforeEach(() => {
+    expect(store.state.currentMediaSource).toMatchObject(
+      DEMO_TEST_MEDIA_SOURCE
+    );
   });
 
-  store.dispatch('setNextTrack');
+  it('clears the queue when setNextTrack is dispatched at the end of the queue', () => {
+    store.replaceState({
+      ...defaultState(),
+      queue: [DEMO_TEST_ITEM_A, DEMO_TEST_ITEM_B],
+      status: PlaybackStatus.playing,
+      currentItemIndex: 1
+    });
 
-  expect(store.state.currentItemIndex).toBe(1);
-  expect(store.state.lastItemIndex).toBe(0);
-});
+    store.dispatch('setNextTrack');
 
-test('When setPreviousTrack is is called, the queue is cleared. Case A', () => {
-  store.replaceState({
-    ...defaultState(),
-    currentTime: 5
+    expect(store.state.status).toBe(PlaybackStatus.stopped);
   });
 
-  store.dispatch('setPreviousTrack');
+  it('increases the item indexes when setNextTrack is dispatched', () => {
+    store.replaceState({
+      ...defaultState(),
+      queue: [DEMO_TEST_ITEM_A, DEMO_TEST_ITEM_B],
+      currentItemIndex: 0
+    });
 
-  expect(store.state.currentTime).toBe(0);
-});
+    store.dispatch('setNextTrack');
 
-test('When setPreviousTrack is is called, the queue is cleared. Case B', () => {
-  store.replaceState({
-    ...defaultState(),
-    currentTime: 1,
-    currentItemIndex: 3
+    expect(store.state.currentItemIndex).toBe(1);
+    expect(store.state.lastItemIndex).toBe(0);
   });
 
-  store.dispatch('setPreviousTrack');
+  it('clears the current time when setPreviousTrack is dispatched', () => {
+    store.replaceState({
+      ...defaultState(),
+      currentTime: 5
+    });
 
-  expect(store.state.currentItemIndex).toBe(2);
-});
+    store.dispatch('setPreviousTrack');
 
-test('When setPreviousTrack is is called, the queue is cleared. Case C', () => {
-  store.replaceState({
-    ...defaultState(),
-    currentTime: 1
+    expect(store.state.currentTime).toBe(0);
   });
 
-  store.dispatch('setPreviousTrack');
+  it('decreases the item indexs when setPreviousTrack is dispatched', () => {
+    store.replaceState({
+      ...defaultState(),
+      currentItemIndex: 3
+    });
 
-  expect(store.state.currentTime).toBe(0);
-});
+    store.dispatch('setPreviousTrack');
 
-test('When resetCurrentTime is is called, the currentTime is set to 0.', () => {
-  store.replaceState({
-    ...defaultState(),
-    currentTime: 1.234
+    expect(store.state.currentItemIndex).toBe(2);
+    expect(store.state.lastItemIndex).toBe(3);
   });
 
-  store.dispatch('resetCurrentTime');
+  it('clears the current time when resetCurrentTime is dispatched', () => {
+    store.replaceState({
+      ...defaultState(),
+      currentTime: 1.234
+    });
 
-  expect(store.state.currentTime).toBe(0);
-});
+    store.dispatch('resetCurrentTime');
 
-test('When resetCurrentItemIndex is is called, the currentItemIndex is set to null.', () => {
-  store.replaceState({
-    ...defaultState(),
-    currentItemIndex: 1
+    expect(store.state.currentTime).toBe(0);
```

Comment:
Use defaultstate?

Reply:
It has to be 0, the default state is null.